sdk/identity/identity/src/util/validateMultiTenant.ts

```diff
@@ -0,0 +1,34 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import { GetTokenOptions } from "@azure/core-auth";
+
+/**
+ * @internal
+ */
+export const multiTenantError = new Error(
+  "Multi-tenant authentication was attempted, but multi-tenant authentication was not enabled in this credential instance."
+);
+
+/**
+ * Verifies whether locally assigned tenants are equal to tenants received through getToken.
+ * Returns the appropriate tenant.
+ * @internal
+ */
+export function processMultiTenantRequest(
+  tenantId?: string,
+  getTokenOptions?: GetTokenOptions
+): string | undefined {
+  if (
+    !getTokenOptions?.allowMultiTenantAuthentication &&
```

Comment:
This should be passed in separately as it's configured on the credential itself not on `GetTokenOptions`

Reply:
For sure! I will do that asap